### PR TITLE
Add --file-path

### PR DESCRIPTION
**Pull request by yingwang-us**
add `-file-path` flag so that SRE team/customers could download component yamls from other locations, e.g. from regional s3 location.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,8 @@
     : namespace to populate razeedeploy resources into (Default 'razeedeploy')
 -s, --file-source=''
     : url that razeedeploy-job should source razeedeploy resource files from (Default 'https://github.com')
+-ft, --file-template=''
```
Comment (alewitt2) [inline README.md:27]:
```suggestion
--ft, --file-template=''
```

**Comment by alewitt2** [comment]
i dont understand why this is necessary. Can you please provide a use case example to show why you cant do what you need with what is currently there

File: src/install.js [hunk 2]
```diff
@@ -104,6 +104,8 @@ async function main() {
     fileSource = fileSource.replace(/\/+$/g, '');
   }
 
+  let fileTemplate = argv['ft'] || argv['file-template'] || 'releases/{{install_version}}/resource.yaml';
```
Comment (alewitt2) [inline src/install.js:107]: make `fileTemplate` a global variable to use throughout the script.

File: src/install.js [hunk 3]
```diff
@@ -166,7 +168,7 @@ async function main() {
         file = yaml.safeLoadAll(file);
         await decomposeFile(file);
         if (autoUpdate) {
-          autoUpdateArray.push({ options: { url: resourceUris[i].uri.replace('{{install_version}}', 'latest/download') } });
+          autoUpdateArray.push({ options: { url: resourceUris[i].uri.replace('{{install_version}}',  argv['file-template'] ? 'latest' : 'latest/download') } });
```
Comment (alewitt2) [inline src/install.js:171]:
```suggestion
          autoUpdateArray.push({ options: { url: resourceUris[i].uri.replace('{{install_version}}',  (argv['ft'] || argv['file-template']) ? 'latest' : 'latest/download') } });
```

File: src/install.js [hunk 4]
```diff
@@ -219,12 +221,16 @@ async function crdRegistered(apiVersion, kind, attempts = 5, backoffInterval = 5
 
 async function download(resourceUriObj) {
   let install_version = (typeof resourceUriObj.install === 'string' && resourceUriObj.install.toLowerCase() !== 'latest') ? `download/${resourceUriObj.install}` : 'latest/download';
+  if (argv['file-template']) {
```
Comment (alewitt2) [inline src/install.js:224]:
```suggestion
  if (argv['ft'] || argv['file-template']) {
```

File: src/install.js [hunk 5]
```diff
@@ -219,12 +221,16 @@ async function crdRegistered(apiVersion, kind, attempts = 5, backoffInterval = 5
...
+    // if file-template is defined, use the version directly
+    install_version = `${resourceUriObj.install}`;
+  }
   let uri = resourceUriObj.uri.replace('{{install_version}}', install_version);
   try {
     log.info(`Downloading ${uri}`);
     return { file: (await axios.get(uri)).data, uri: uri };
   } catch (e) {
-    let latestUri = resourceUriObj.uri.replace('{{install_version}}', 'latest/download');
+    let latestUri = resourceUriObj.uri.replace('{{install_version}}', argv['file-template'] ? 'latest' : 'latest/download');
```
Comment (alewitt2) [inline src/install.js:233]:
```suggestion
    let latestUri = resourceUriObj.uri.replace('{{install_version}}', (argv['ft'] || argv['file-template']) ? 'latest' : 'latest/download');
```

File: README.md (hunk 1 above)
Comment (alewitt2) [reply README.md:27]: also need to add this to the help menu in src/install.js `if (argv.h || argv.help) { ...`

File: src/install.js (hunk 2 above)
Comment (alewitt2) [inline src/install.js:107]: lets change the arg name to:
```suggestion
  let filePath = argv['fp'] || argv['file-path'] || 'releases/{{install_version}}/resource.yaml';
```

File: README.md (hunk 1 above)
Comment (yingwang-us) [reply README.md]: fixed

**Pull request merged**